app/services/katello/pulp3/rpm.rb

```diff
@@ -26,41 +26,13 @@ def self.page_options(page_opts = {})
       end
 
       def requires
-        results = []
-        flags = {'GT' => '>', 'LT' => '>', 'EQ' => '=', 'GE' => '>=', 'LE' => '<='}
-
-        backend_data['requires']&.each do |requirement|
-          requires_str = ""
-          if requirement.count < 3
-            requires_str = requirement.first
-            results << requires_str
-          else
-            requirement[1] = flags[requirement[1]]
-            requirement[0...2].each { |requirement_piece| requires_str += "#{requirement_piece} " }
-            requirement[2...-1].each { |requirement_piece| requires_str += "#{requirement_piece}." }
-            results << requires_str[0...-1]
-          end
-        end
-        results.uniq
+        results = Util::Package.parse_dependencies(backend_data['requires'])
+        results
       end
 
       def provides
-        results = []
-        flags = {'GT' => '>', 'LT' => '>', 'EQ' => '=', 'GE' => '>=', 'LE' => '<='}
-
-        backend_data['provides']&.each do |provided|
-          provides_str = ""
-          if provided.count < 3
-            provides_str = provided.first
-            results << provides_str
-          else
-            provided[1] = flags[provided[1]]
-            provided[0...2].each { |provided_piece| provides_str += "#{provided_piece} " }
-            provided[2...-1].each { |provided_piece| provides_str += "#{provided_piece}." }
-            results << provides_str[0...-1]
-          end
-        end
-        results.uniq
+        results = Util::Package.parse_dependencies(backend_data['provides'])
+        results
```

Comment:
no need to grab these into a variable and return them, you can just call `Util::Package.parse_dependencies(backend_data['provides'])`

same for the requires method too

Reply:
updated!